_health-care/_js/_components/health-care-app.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import _ from 'lodash';
+import lodashDeep from 'lodash-deep';
+
+import IntroductionPanel from './introduction-panel.jsx';
+import Nav from './nav.jsx';
+
+// Add deep object manipulation routines to lodash.
+_.mixin(lodashDeep);
+
+class HealthCareApp extends React.Component {
+  constructor() {
+    super();
+    this.publishStateChange = this.publishStateChange.bind(this);
+    this.state = {
+      applicationData: {
+        introduction: {
+          fullName: {
+            first: "William",
+            middle: "",
+            last: "Shakespeare",
+            suffix: "",
+          },
+
+          happyDate: {
+            month: "1",
+            day: "15",
+            year: "1997",
+          },
+
+          mothersMaidenName: "Arden",
+
+          sadDate: {
+            month: "4",
+            day: "5",
+            year: "2014",
+          }
+        }
+      }
+    };
+
+    // Merge in saved data.
+    // TODO(awong): If a field is ever removed in a new version of the code,
+    // this will preserve the key. The result of JSON.parse() should first
+    // be filtered to only contain keys also in this.state.applicationData.
+    const savedDataJson = window.localStorage["health-app-data"];
+    if (savedDataJson !== undefined) {
+      const savedData = JSON.parse(savedDataJson);
+      this.state.applicationData = _.deepMapValues(
```

Comment:
Oh, because `this.state.applicationData` goes away if the browser window is closed but `window.localStorage` persists? So in that case we would want to update `this.state.applicationData` with their saved data... got it :)